File: pset1/readability/readability.c
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#include <math.h>
#include <stdio.h>

int count_letters(string text)
{
    int numLetters = 0;
    for (int i = 0; text[i] != '\0'; i += 1)
    {
        if (isalnum(text[i]) != 0)
        {

            numLetters += 1;
        }
    }

    return numLetters;
}

int number_of_words(string text)
{
    char asp = (int)39;
    int number_words = 0;
    for (int i = 0; text[i] != '\0'; i += 1)
    {
        int newIndex = i;
        if (text[i + 1] != '\0')
        {
            newIndex = i + 1;
        }
        if (isalnum(text[i]) == 0 && text[i] != '-' && isalnum(text[newIndex]) != 0 && (int)text[newIndex] != 39)
        {
            number_words += 1;

            if (text[i] == asp)
            {
                number_words -= 1;

            }
        }
        else if (text[i + 1] == '\0')
        {
            number_words += 1;

        }

    }
    return number_words;
}

int count_sentence(string text)
{
    int numbers_sentence = 0;
    for (int i = 0; text[i] != '\0'; i += 1)
    {
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            numbers_sentence += 1;
        }
    }
    return numbers_sentence;
}

int main(void)
{
    string text = get_string("Set text: \n");
    printf("%i\n", isalnum(text[0]));
    int num_letters = count_letters(text);
    int num_words = number_of_words(text);
    int num_sentence = count_sentence(text);
    float l = 0.0588  * (float) num_letters / num_words * 100;
    float s = 0.296 * (float) num_sentence / num_words * 100;
    float index = round(l - s - 15.8);
    if (index <= 1)
    {
        printf("Before Grade %.0f\n", index);
    }
    else if (index > 1 && index <= 15)
    {
        printf("Grade %.0f\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}

File: readability_py/readability.py
import re


def count_letters(text):
    text = text.lower()
    rep = [' ', '.', ',', '!', '?', ';', "'", '"', ':', '-', '`']
    for l in range(len(rep)):
        text = text.replace(rep[l], '')
    return len(text)


def count_words(text):
    text = text.lower()
    rep = ['.', ',', '!', '?', ';', "'", '"', ':']
    for l in range(len(rep)):
        text = text.replace(rep[l], '')
    number_words = text.split()
    return len(number_words)


def count_sentence(text):
    text = text.lower()
    rep = [',', ';', "'", '"', ':', '`', '-']
    for l in range(len(rep)):
        text = text.replace(rep[l], '')
    text = text[:-1]
    number_sentence = re.split('[?.!]', text)
    return len(number_sentence)


text = input('Text: ')
number_l = count_letters(text)
number_w = count_words(text)
number_s = count_sentence(text)

l = 0.0588 * number_l / number_w * 100
s = 0.296 * number_s / number_w * 100
index = round(l - s - 15.8)

if index <= 1:
    print(f'Before Grade {index}')
elif index > 1 and index <= 15:
    print(f'Grade {index}')
elif index >= 16:
    print(f'Grade 16+')